### DISCUSSION TITLE
Running local error

### DISCUSSION THREAD

**Discussion by satyammayy** [phaticusthiccy/WhatsAsenaDuplicated#229] [Q&A]
Please help

**Comment by satyammayy** [comment]
> 

ASENA_SESSION="ASENA;;;eyJjbxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
ANTI_LINK="false"
ANTILINK_MESSAGE="default"
AUTO_BIO="false"
PROXY="false"
GAN_IMAGE="https://i.hizliresim.com/loUtAb.jpg"
LANGUAGE="EN" 
ALIVE_MESSAGE="default"
KICKME_MESSAGE="default"
BLOCK_CHAT=false
ADD_MESSAGE="default"
MUTE_MESSAGE="default"
NO_LOG="false"
FULL_EVA="false"
BLOCK_MESSAGE="default"
UNBLOCK_MESSAGE="default"
UNMUTE_MESSAGE="default"
WORK_TYPE="public"  
PROMOTE_MESSAGE="default"
DEMOTE_MESSAGE="default"
BAN_MESSAGE="default"
AFK_MESSAGE="default"
HANDLERS="^[.!;]"
SEND_READ=false
HEROKU=false
HEROKU_API_KEY="" # Replace with your Heroku API key
HEROKU_APP_NAME="" # Replace with your Heroku app name
REMOVE_BG_API_KEY=false # Replace with your remove bg API key "key"
DEEPAI_API_KEY=false # Replace with your DeepAI API key "key"
NO_ONLINE=true
SUDO=false








my config.env

**Comment by satyammayy** [comment]
PS C:\Users\Laiphadibi\project\WhatsAsenaDuplicated> node bot.js
You entered an invalid language. English language was chosen.
node:internal/errors:464
    ErrorCaptureStackTrace(err);
    ^

TypeError [ERR_UNESCAPED_CHARACTERS]: Request path contains unescaped characters
    at new NodeError (node:internal/errors:371:5)
    at new ClientRequest (node:_http_client:154:13)
    at Object.request (node:https:353:10)
    at Request.request (C:\Users\Laiphadibi\project\WhatsAsenaDuplicated\node_modules\heroku-client\lib\request.js:113:19)
    at Heroku.request (C:\Users\Laiphadibi\project\WhatsAsenaDuplicated\node_modules\heroku-client\lib\index.js:14:20)
    at Heroku.get (C:\Users\Laiphadibi\project\WhatsAsenaDuplicated\node_modules\heroku-client\lib\index.js:18:17)
    at C:\Users\Laiphadibi\project\WhatsAsenaDuplicated\bot.js:140:30
    at ChildProcess.exithandler (node:child_process:404:5)
    at ChildProcess.emit (node:events:390:28)
    at maybeClose (node:internal/child_process:1064:16) {
  code: 'ERR_UNESCAPED_CHARACTERS'
}











the errrorrrrr

**Comment by phaticusthiccy** [comment]
Please dont use  `#` and dont leave blank sections under HEROKU module.

**Reply by satyammayy** [reply]
PS C:\Users\Laiphadibi\project\WhatsAsenaDuplicated> node bot.js
Loading EN language...
C:\Users\Laiphadibi\project\WhatsAsenaDuplicated\bot.js:141
                    forID = formation[0].id;
                                         ^

TypeError: Cannot read properties of undefined (reading 'id')
    at C:\Users\Laiphadibi\project\WhatsAsenaDuplicated\bot.js:141:42
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async C:\Users\Laiphadibi\project\WhatsAsenaDuplicated\bot.js:140:17

**Reply by satyammayy** [reply]
ASENA_SESSION="ASENA;;;eyJjbGllbnRJ"
ANTI_LINK=false
ANTILINK_MESSAGE=default
AUTO_BIO=false
PROXY=false
GAN_IMAGE=https://i.hizliresim.com/loUtAb.jpg
LANGUAGE=EN
ALIVE_MESSAGE=default
KICKME_MESSAGE=default
BLOCK_CHAT=false
ADD_MESSAGE=default
MUTE_MESSAGE=default
NO_LOG=false
FULL_EVA=false
BLOCK_MESSAGE=default
UNBLOCK_MESSAGE=default
UNMUTE_MESSAGE=default
WORK_TYPE=public  
PROMOTE_MESSAGE=default
DEMOTE_MESSAGE=default
BAN_MESSAGE=default
AFK_MESSAGE=default
HANDLERS=^[.!;]
SEND_READ=false
HEROKU=false
HEROKU_API_KEY= 232ab8b6-3855-4054-ba00-026b950de400
HEROKU_APP_NAME=asdasdadasdasdasdasd
REMOVE_BG_API_KEY=false
DEEPAI_API_KEY=false
NO_ONLINE=true
SUDO=false




.env

**Comment by phaticusthiccy** [comment]
Did you changed some codes?

**Reply by satyammayy** [reply]
None

**Reply by satyammayy** [reply]
i just cloned it and make changes to env  and simply node bot.js

**Reply by boenmay123** [reply]
HE MUST HAVE COPIED THE SOURCE CODE FROM OTHER DEVELOPERS THAT'S WHY HE DOESNT KNOW MUCH ABOUT RUNNING LOCAL

**Reply by phaticusthiccy** [reply]
Firstly, cut the grap @boenmay123 
@satyammayy Please, re-clone the repository and don't make any changes on local codes.

**Reply by satyammayy** [reply]
I did reclone it  and cant make it to run it throws same error
